crates/bifrost/src/lib.rs

```diff
@@ -47,3 +49,23 @@ pub(crate) fn create_static_metadata(opts: &Options, num_partitions: u64) -> Log
 
     Logs::new(Version::MIN, log_chain)
 }
+
+task_local! {
+    static BIFROST: Bifrost;
+}
+
+/// Sets the bifrost handle as a task local variable so that the future can access Bifrost more easily.
+pub fn with_bifrost<F>(future: F, bifrost: Bifrost) -> impl Future<Output = anyhow::Result<()>>
```

Comment:
Yes, you are right. I will remove this bit and instead try to pass Bifrost along explicitly because if it is not a common dependency (similar to metadata), then it becomes a nuisance propagating it implicitly via `with_bifrost`.